baseapp/baseapp.go

```diff
@@ -602,6 +623,12 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte, tx sdk.Tx) (result sdk
 	ctx := app.getContextForAnte(mode, txBytes)
 	ctx = app.initializeContext(ctx, mode)
 
+	// only run the tx if there is block gas remaining
+	if mode == runTxModeDeliver && ctx.BlockGasMeter().PastLimit() {
+		result = sdk.ErrOutOfGas("no block gas left to run tx").Result()
```

Comment:
Should we have a dedicated ABCI code for out of block gas, or not?

Reply:
I think we should.

Reply:
OKAY so here is the thing - _unless_ we want to define a new type of gas meter in  `types/gas.go` then the first time this meter fails it's going to fail when the gas is being deducted from after the transaction has been run (later in this function). Which now means that the block gas meter is going to be failing with two different ABCI codes which doesn't make any sense - so really our options are either:
 - introduce a new ABCI error code which requires:
    - dupe code for a new gas meter OR maybe modification of initialization of the gas meter to include an error code as input
    - adding an extra catch in the recover statement within `runTx` 
 - OR we can just be okay looking at the error log to indicate it's a block gas error
______________
Probably good to make the switch to a new error code I think maybe hahaha.

Reply:
cwgoes [11:14 AM]
yes, that's annoying
I'm OK with either, no strong opinion

rige [11:16 AM]
yeah I mean to me the only crappy thing is having an extra switch in the recover statement - not a big deal to implement now that I’ve thought about it - just wondering if we actually want it

cwgoes [11:16 AM]
I don't think it matters that much
this should only happen with a bad proposer

rige [11:17 AM]
fair then I move that we leave it just to keep the code a bit simpler